### PR TITLE
Allow drag and drop or copy and paste of properties in the extension editor

### PR DESCRIPTION
**Pull request by D8H**
https://gdevelop-storybook.s3.amazonaws.com/property-editor-refresh/latest/index.html?path=/story/eventsbasedbehavioreditor-eventsbasedbehavioreditordialog--default

### Changes
- It reports the new features that was added to the effect editor.
- drag and drop
- copy and paste
- placeholder view for empty list
- Title and content indentation
- "Add" button always visible
- Fix the "visibility" checkbox layout for small screens

### Wins
- Users no longer have to write easing function choice properties by hand (or edit the project with a text editor)
- Beginners have access to the "property" section of the doc directly
- Users can move visible properties on the top more easily

### CODE REVIEW

File: newIDE/app/src/EventsBasedBehaviorEditor/EventsBasedBehaviorPropertiesEditor.js [hunk 1]
```diff
@@ -22,15 +21,63 @@ import StringArrayEditor from '../StringArrayEditor';
...
+    alignItems: 'center',
+  },
+};
+
+export const useEffectOverridingAlertDialog = () => {
+  const { showConfirmation } = useAlertDialog();
+  return async (existingEffectNames: Array<string>): Promise<boolean> => {
+    return await showConfirmation({
+      title: t`Existing effects`,
+      message: t`These effects are already exists:${'\n\n - ' +
```
Comment (4ian) [inline newIDE/app/src/EventsBasedBehaviorEditor/EventsBasedBehaviorPropertiesEditor.js:72]:
```suggestion
      message: t`These effects already exists:${'\n\n - ' +
```

File: newIDE/app/src/EventsBasedBehaviorEditor/EventsBasedBehaviorPropertiesEditor.js [hunk 2]
```diff
@@ -89,420 +136,746 @@ const getExtraInfoArray = (property: gdNamedPropertyDescriptor) => {
...
-      groupNames.add(group);
-    }
-    return [...groupNames].sort((a, b) => a.localeCompare(b));
-  };
+  const copyProperty = React.useCallback(
+    (property: gdNamedPropertyDescriptor) => {
+      Clipboard.set(PROPERTIES_CLIPBOARD_KIND, [
+        {
+          name: property.getName(),
+          serializedEffect: serializeToJSObject(property),
```
Comment (4ian) [inline newIDE/app/src/EventsBasedBehaviorEditor/EventsBasedBehaviorPropertiesEditor.js:199]:
```suggestion
          serializedProperty: serializeToJSObject(property),
```

File: newIDE/app/src/EventsBasedBehaviorEditor/EventsBasedBehaviorPropertiesEditor.js [hunk 3]
```diff
@@ -89,420 +136,746 @@ const getExtraInfoArray = (property: gdNamedPropertyDescriptor) => {
...
+      }> = [];
+      const existingNamedProperties: Array<{
+        name: string,
+        serializedProperty: string,
+      }> = [];
+      propertyContents.forEach(effectContent => {
+        const name = SafeExtractor.extractStringProperty(effectContent, 'name');
+        const serializedProperty = SafeExtractor.extractObjectProperty(
+          effectContent,
+          'serializedEffect'
```
Comment (4ian) [inline newIDE/app/src/EventsBasedBehaviorEditor/EventsBasedBehaviorPropertiesEditor.js:225]: Same here

File: newIDE/app/src/EventsBasedBehaviorEditor/EventsBasedBehaviorPropertiesEditor.js [hunk 4]
```diff
@@ -89,420 +136,746 @@ const getExtraInfoArray = (property: gdNamedPropertyDescriptor) => {
...
+
+        if (properties.has(name)) {
+          existingNamedProperties.push({ name, serializedProperty });
+        } else {
+          newNamedProperties.push({ name, serializedProperty });
+        }
+      });
+
+      let firstAddedPropertyName: string | null = null;
+      let index = effectInsertionIndex;
```
Comment (4ian) [inline newIDE/app/src/EventsBasedBehaviorEditor/EventsBasedBehaviorPropertiesEditor.js:239]:
```suggestion
      let index = propertyInsertionIndex;
```

File: newIDE/app/src/EventsBasedBehaviorEditor/EventsBasedBehaviorPropertiesEditor.js [hunk 5]
```diff
@@ -89,420 +136,746 @@ const getExtraInfoArray = (property: gdNamedPropertyDescriptor) => {
...
-                            disableAlpha
-                            fullWidth
-                            color={property.getValue()}
-                            onChange={color => {
-                              property.setValue(color);
-                              this.forceUpdate();
-                              this.props.onPropertiesUpdated &&
-                                this.props.onPropertiesUpdated();
+  const pasteProperties = React.useCallback(
+    async effectInsertionIndex => {
```
Comment (4ian) [inline newIDE/app/src/EventsBasedBehaviorEditor/EventsBasedBehaviorPropertiesEditor.js:208]: name to update

**Review by 4ian — APPROVED** [review]

File: newIDE/app/src/EventsBasedBehaviorEditor/EventsBasedBehaviorPropertiesEditor.js (hunk 5 above)
Comment (D8H) [reply newIDE/app/src/EventsBasedBehaviorEditor/EventsBasedBehaviorPropertiesEditor.js]: Sorry, I though I did a pass on this. I must have been distracted.

**Pull request merged**